src/AWS.Deploy.CLI/Commands/DeployCommand.cs

```diff
@@ -130,6 +130,19 @@ public async Task ExecuteAsync(string applicationName, string deploymentProjectP
 
             await CreateDeploymentBundle(orchestrator, selectedRecommendation, cloudApplication);
 
+            if (!string.IsNullOrEmpty(saveSettingsFilePath))
+            {
+                var errorMessage = await _deploymentSettingsHandler.SaveSettings(saveSettingsFilePath, selectedRecommendation, cloudApplication, _session);
+                if (!string.IsNullOrEmpty(errorMessage))
+                {
+                    _toolInteractiveService.WriteErrorLine(errorMessage);

```

Comment:
I am not sure if we want to fail the entire deployment in we fail to save the settings file. For now I am wrapping `_deploymentSettingsHandler.SaveSettings(..)` in a `try-catch` and logging the error message in the catch block. Interested in @normj's and @ashovlin's opinion on it.

Reply:
I was also wondering about the ordering of `SaveSettings` vs. the deployment itself. Do we want to be saving the settings before we know whether or not they'll be successful?

On Phil's original question, I lean towards failing the deployment to avoid anything unexpected:
1. User deploys settings A
2. User deploys settings B, but save fails so file is still at A
3. User deploys again, thinking they are applying B, but really applying A

Reply:
Fixed this to fail the deployment if the operation to save the settings file fails.

Reply:
To Alex's comment, Currently, the settings file is created just before the deployment starts. At this point all option settings have been validated and the deployment bundle has been created. If the deployment fails, then chances are low that it happened due to a user error. I think it would be valuable to save the settings file for diagnostic purposes even if the deployment fails.